Roles/Core/RoleBase.cs

```diff
@@ -1,33 +1,57 @@
 ﻿using AmongUs.GameOptions;
 using Hazel;
 using InnerNet;
+using TOHE.Roles.Core;
 using UnityEngine;
 
 namespace TOHE;
 
 public abstract class RoleBase
 {
+    public PlayerState _state;
+#pragma warning disable IDE1006
+    public PlayerControl _Player => Utils.GetPlayerById(_state.PlayerId);
+    public List<byte> _playerIdList => Main.PlayerStates.Values.Where(x => x.MainRole == _state.MainRole).Select(x => x.PlayerId) as List<byte>;
+#pragma warning restore IDE1006
+
+    public float AbilityLimit { get; set; } = -100;
+    public virtual bool IsEnable { get; set; } = false;
+    public void OnInit() // CustomRoleManager.RoleClass executes this
+    {
+        IsEnable = false;
+        Init();
+    }
+
+    public void OnAdd(byte playerid) // The player with the class executes this
+    {
+        _state = Main.PlayerStates.Values.FirstOrDefault(state => state.PlayerId == playerid);
+        try { 
+            CustomRoleManager.RoleClass.FirstOrDefault(r => r.Key == _state.MainRole).Value.IsEnable = true;
+            this.IsEnable = true; // Not supposed to be used, but some methods may have still implemented that check.
+        } catch { } // temporary try catch
+
```

Comment:
okay it's ready, but will have to change this once all the current role pr's are merged (it's temporarily in a try catch bcuz of that)